### DISCUSSION TITLE
Client credentials flow

### DISCUSSION THREAD

**Discussion by itsmesuniljacob** [ory/hydra#2705] [Q&A] [answered]
Hi Team,

Just wanted to check if ORY Hydra is the right tool for micro service to service authentication. The scenario is as below:
frontend app needs talks to authorisation server and get the access_token ( Jwt token ) and pass the token to the back end app and the backend app validates the token with the auth server to allow/deny the same. 

We will be having diff environments ( dev/stage/prod ). Can I create multiple audiences for the same and have `client_id` and `client_secrets` for the same?

**Comment by Benehiko** [comment, marked as answer]
Hi @xylene314 

Could you maybe expand on your use case. From what you've described, you requires [Kratos](https://www.ory.sh/kratos/) - an Identity Management server. Kratos gives a session cookie/token for the frontend which can be validated by your backend service middleware or through [Oathkeeper](https://www.ory.sh/oathkeeper/).

**Reply by vinckr** [reply]
Hey @itsmesuniljacob  ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
